src/main/java/core/basesyntax/service/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private String report = "";
+    private FileReader fileReader = new FileReaderImpl();
+    private FileWriter fileWriter = new FileWriterImpl();
+    private CalculateBalance calculateBalance = new CalculateBalanceImpl();
+
+    @Override
+    public void makeReport() {
+        calculateBalance.calculateBalance(fileReader.readFromFileToArray());
+
+        for (String element: Storage.result.keySet()) {
+            report += (element + ", " + Storage.result.get(element) + System.lineSeparator());
+        }
+        fileWriter.writeToFile(report);
```

Comment:
This is basically the whole flow of the app - reading from file, parsing to models, applying with Strategy pattern, generating report, writing it to a file. But since this is a ReportGeneration class, you should leave only the generating report part here. Put other logic into Main class, or Create the high level FruitShopService class